src/FSharp.Core/array.fs

```diff
@@ -1934,6 +1934,46 @@ module Array =
     module Parallel =
         open System.Threading.Tasks
 
+        [<CompiledName("TryFindIndex")>]
+        let tryFindIndex predicate (array: _[]) =
+            checkNonNull "array" array
+
+            let pResult =
+                Parallel.For(
+                    0,
+                    array.Length,
+                    (fun i pState ->
+                        if predicate array[i] then
+                            pState.Break())
+                )
+
+            pResult.LowestBreakIteration |> Option.ofNullable |> Option.map int
+
+        [<CompiledName("TryFind")>]
+        let tryFind predicate (array: _[]) =
+            array |> tryFindIndex predicate |> Option.map (fun i -> array[i])
```

Comment:
Not that it would matter most of the time (especially since this is `Array.Parallel` and not regular `Array`), but given that this is standard library code that probably won't change very often, it might not hurt to avoid the extra allocations here[^1][^2][^3]:

```suggestion
        let inline tryFindIndexAux predicate (array: _ array) =
            checkNonNull (nameof array) array
            
            let pResult =
                Parallel.For(
                    0,
                    array.Length,
                    (fun i pState ->
                        if predicate array[i] then
                            pState.Break())
                )

            pResult.LowestBreakIteration

        [<CompiledName("TryFindIndex")>]
        let tryFindIndex predicate (array: _ array) =
            let i = tryFindIndexAux predicate array
            if i.HasValue then Some (int (i.GetValueOrDefault()))
            else None

        [<CompiledName("TryFind")>]
        let tryFind predicate (array: _ array) =
            let i = tryFindIndexAux predicate array
            if i.HasValue then Some (array[int (i.GetValueOrDefault()])
            else None
```

[^1]: 2 for `tryFindIndex`: one `Some` from `Option.ofNullable` and another from `Option.map`.
[^2]: 3 for `tryFind`: one `Some` from `Option.ofNullable`, another from the first `Option.map`, and a third from the outer `Option.map`.
[^3]: Or maybe I've just read too many Stephen Toub PRs 🙃.